src/main/java/org/truffleruby/language/methods/CallInternalMethodNode.java

```diff
@@ -62,13 +62,15 @@ protected Object callCached(Object[] rubyArgs,
             @Cached("method.getCallTarget()") RootCallTarget cachedCallTarget,
             @Cached("method") InternalMethod cachedMethod,
             @Cached("createCall(cachedMethod.getName(), cachedCallTarget)") DirectCallNode callNode) {
+        assert RubyArguments.assertArguments(rubyArgs);
         return callNode.call(rubyArgs);
     }
 
     @Specialization(guards = "!method.alwaysInlined()", replaces = "callCached")
     protected Object callUncached(Object[] rubyArgs,
             @Bind("getMethod(rubyArgs)") InternalMethod method,
             @Cached IndirectCallNode indirectCallNode) {
+        assert RubyArguments.assertArguments(rubyArgs);
```

Comment:
Need to assert in all specializations, including the ones below, or assert before the CallInternalMethodNode like it was done.
I think before is better and will incur far less conflicts with ongoing work in this node and DispatchNode.

Reply:
`CallInternalMethodNode` is used in many places - if we put the assert there we cover all uses.

Reply:
Yeah I was also hesitating, but it's only used in 5 places and there is already a JavaDoc comment explaining this requirement on CallInternalMethodNode#execute, so I think better as it is.
Concretely, I don't think we want to deal with the conflict needlessly here.